### PR TITLE
Replace VM2 with isolated-vm

### PR DESCRIPTION
**Pull request by techcoderx**
Due to https://github.com/patriksimek/vm2/issues/533, the execution engine will have to be replaced with [isolated-vm](https://github.com/laverdet/isolated-vm).

The replacement package allows for setting a memory limit (in MB). This may be useful to define a "memory limit" on a transaction basis to calculate a max tx fee for contract calls.

Tested working on my node so far with some contract calls.

### CODE REVIEW

**Comment by vaultec81** [comment]
@techcoderx Can you pull from master & resolve merge conflicts?

**Comment by techcoderx** [comment]
Done, please double check to make sure all the latest methods are being passed in correctly.

File: src/services/contractEngine.ts [hunk 1]
```diff
@@ -426,71 +495,96 @@ export class ContractEngine {
...
+        return new ivm.ExternalCopy(result)
+      }))
+      context.global.setSync('state_del', new ivm.Reference(async (key: string) => {
+        await state.client.del(key)
+      }))
+      context.global.setSync('done', () => {
+        stateMerkle = state.finish().stateMerkle
+      })
+      const compiled = await isolate.compileScript(code)
+      await compiled.run(context)
```
Comment (vaultec81) [inline src/services/contractEngine.ts:587]: Stalls and fails to execute

**Review by vaultec81 — CHANGES_REQUESTED** [review]
Contract execution stalls and never completes on my end. Doesn't produce an output or any meaningful output

**Comment by techcoderx** [comment]
Apologies for the delay. Missed a few commas in the template which was why it did not compile.

**Review by vaultec81 — APPROVED** [review]
LGTM - passes tests

**Pull request merged**